src/EVM/SymExec.hs

```diff
@@ -568,27 +554,29 @@ verifyContract :: forall m . App m
   -> [String]
   -> VeriOpts
   -> Maybe (Precondition RealWorld)
-  -> Maybe (Postcondition RealWorld)
+  -> Postcondition RealWorld
   -> m (Expr End, [VerifyResult])
-verifyContract solvers theCode signature' concreteArgs opts maybepre maybepost = do
-  verifyContractWithSession solvers Nothing theCode signature' concreteArgs opts maybepre maybepost
+verifyContract solvers theCode signature' concreteArgs opts maybepre post = do
+  calldata <- mkCalldata signature' concreteArgs
+  preState <- liftIO $ stToIO $ abstractVM calldata theCode maybepre False
+  let fetcher = Fetch.oracle solvers Nothing opts.rpcInfo
+  verify solvers fetcher opts preState post
 
 -- Used only in testing
-verifyContractWithSession :: forall m . App m
+exploreContract :: forall m . App m
   => SolverGroup
-  -> Maybe Fetch.Session
   -> ByteString
   -> Maybe Sig
   -> [String]
   -> VeriOpts
   -> Maybe (Precondition RealWorld)
-  -> Maybe (Postcondition RealWorld)
-  -> m (Expr End, [VerifyResult])
-verifyContractWithSession solvers sess theCode signature' concreteArgs opts maybepre maybepost = do
+  -> m (Expr End)
+exploreContract solvers theCode signature' concreteArgs opts maybepre = do
   calldata <- mkCalldata signature' concreteArgs
   preState <- liftIO $ stToIO $ abstractVM calldata theCode maybepre False
-  let fetcher = Fetch.oracle solvers sess opts.rpcInfo
-  verify solvers fetcher opts preState maybepost
+  let fetcher = Fetch.oracle solvers Nothing opts.rpcInfo
+  executeVM fetcher opts.iterConf preState
+
```

Comment:
this is only used in one test. It will also need to be removed when we return an `[Expr  End]`, rather than an `Expr End` with a bunch of ITE's in it. So we can also remove it now, if you like, along with the test.